infrastructure/src/main/java/org/corfudb/infrastructure/BaseServer.java

```diff
@@ -29,40 +38,65 @@
 
     /** HandlerMethod for the base server. */
     @Getter
+    @Deprecated
     private final HandlerMethods handler = HandlerMethods.generateHandler(MethodHandles.lookup(), this);
 
+    /**
+     * RequestHandlerMethods for the Base server
+     */
+    @Getter
+    private final RequestHandlerMethods handlerMethods =
+            RequestHandlerMethods.generateHandler(MethodHandles.lookup(), this);
+
     @Override
+    @Deprecated
     public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
         return getState() == ServerState.READY;
     }
 
+    @Override
+    public boolean isServerReadyToHandleReq(RequestMsg request) {
+        return getState() == ServerState.READY;
+    }
+
     public BaseServer(@Nonnull ServerContext context) {
         this.serverContext = context;
         executor = Executors.newFixedThreadPool(serverContext.getBaseServerThreadCount(),
                 new ServerThreadFactory("baseServer-", new ServerThreadFactory.ExceptionHandler()));
     }
 
     @Override
+    @Deprecated
     protected void processRequest(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
         executor.submit(() -> getHandler().handle(msg, ctx, r));
     }
 
+    @Override
+    protected void processRequest(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        executor.submit(() -> getHandlerMethods().handle(req, ctx, r));
+    }
+
     @Override
     public void shutdown() {
         super.shutdown();
         executor.shutdown();
     }
 
     /**
-     * Respond to a ping message.
+     * Respond to a ping request.
      *
-     * @param msg   The incoming message
-     * @param ctx   The channel context
+     * @param req   The incoming request message.
+     * @param ctx   The channel context.
      * @param r     The server router.
      */
-    @ServerHandler(type = CorfuMsgType.PING)
-    private void ping(CorfuMsg msg, ChannelHandlerContext ctx, IServerRouter r) {
-        r.sendResponse(ctx, msg, CorfuMsgType.PONG.msg());
+    @RequestHandler(type = RequestPayloadMsg.PayloadCase.PING_REQUEST)
+    private void handlePing(RequestMsg req, ChannelHandlerContext ctx, IServerRouter r) {
+        log.info("handlePing[{}]: Ping message received from {} {}", req.getHeader().getRequestId(),
```

Comment:
Info is too verbose for a ping. Change you change info to `debug` level.

Reply:
Done.